ai21/clients/studio/resources/studio_resource.py

```diff
@@ -18,10 +18,11 @@ def _cast_response(
     response_cls: Optional[ResponseT],
     stream_cls: Optional[AsyncStreamT] = None,
     stream: bool = False,
+    streaming_decoder: Optional[Any] = None,
```

Comment:
Let's create an interface called `SSEDecoder` which exposes 
-  `async def aiter(self, response: httpx.Response):`
-  `def iter(self, response: httpx.Response):`
 and have both our classes implement it, so we won't use `Any`